### PR TITLE
upgrade to Yarn 4 and clean up some deps

### PR DESCRIPTION
**Pull request by turadg**
Lots of cleanup of dependencies. Principally:

- Upgrading Yarn to v4
- Adding a script to update all deps to match an NPM tag

### CODE REVIEW

File: _agstate/agoric-servers/package.json [hunk 1]
```diff
@@ -14,6 +14,6 @@
   "author": "Agoric",
   "license": "Apache-2.0",
   "dependencies": {
-    "@agoric/cosmic-swingset": "^0.41.3"
+    "@agoric/cosmic-swingset": "^0.42.0-u16.2"
```
Comment (dckc) [inline _agstate/agoric-servers/package.json:17]: We shouldn't have to maintain `_agstate` like this. Marking this one as tech debt:

 - https://github.com/Agoric/agoric-sdk/issues/8817

**Review by dckc — APPROVED** [review]
I could use more docs on the script, but I'm willing to bet it's a lot better than the status quo, so I'm happy if they come after this PR.

File: contract/src/offer-up.contract.js [hunk 2]
```diff
@@ -59,8 +59,10 @@ export const meta = {
     { maxItems: M.bigint() },
   ),
 };
+harden(meta);
```
[the remark this answers is not in the document]
Comment (turadg) [reply contract/src/offer-up.contract.js:62]: by the updated Endo lint config

File: scripts/update-dependencies.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env node
```
[the remark this answers is not in the document]
Comment (turadg) [reply scripts/update-dependencies.js:1]: I'll add a note.

File: package.json [hunk 4]
```diff
@@ -3,35 +3,38 @@
...
   "useWorkspaces": true,
   "workspaces": [
     "contract",
     "ui"
   ],
   "resolutions-note": "work-around for https://github.com/Agoric/agoric-sdk/issues/8621",
   "resolutions": {
-    "ses": "1.3.0",
-    "@endo/bundle-source": "2.5.2-upstream-rollup",
-    "@endo/captp": "3.1.1",
```
[the remark this answers is not in the document]
Comment (turadg) [reply package.json:15]: maybe this looked like the resolutions are gone, but they're just updated. I hope to obviate them eventually but didn't manage yet

[the remark this answers is not in the document]
Comment (dckc) [reply package.json:15]: oops. reading too fast.

File: scripts/update-dependencies.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (dckc) [reply scripts/update-dependencies.js:1]: An entry in the `scripts` key of `package.json` would make it more discoverable, too. `yarn fix-deps`?

[the remark this answers is not in the document]
Comment (turadg) [reply scripts/update-dependencies.js:1]: for now it's a "if you know you know" kind of tool. I think we need some design discussion around the UX to solve this problem consistently across repos. It should be something you can run with the `agoric` command or an `npx` so you don't need the script copied everywhere.

**Pull request merged**